velox/core/QueryCtx.cpp

```diff
@@ -79,4 +79,16 @@ QueryCtx::QueryCtx(
   return fmt::format("query.{}.{}", queryId.c_str(), seqNum++);
 }
 
+void QueryCtx::checkNumSpillBytes(uint64_t bytes) {
+  if (queryConfig_.maxSpillBytes() > 0) {
+    const auto numSpillBytes = numSpillBytes_.fetch_add(bytes) + bytes;
+    VELOX_USER_CHECK_LE(
```

Comment:
See exceededPerQueryLocalLimit example in java and use dedicated error code as well instead of a generic velox throw. thanks!

Reply:
CMIIW I do not find ErrorCode utilities in Velox, do you mean that we need to add a subclass of VeloxUserError?